tensorflow/python/ops/tensor_forest_ops.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Ops for tensor_forest."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from tensorflow.python.ops import resources
+
+from tensorflow.python import ops
+from tensorflow.python.ops import gen_tensor_forest_ops
+from tensorflow.python.training import saver
+
+
+class VariableSavable(saver.BaseSaverBuilder.SaveableObject):
```

Comment:
TreeVariableSaveable.

Reply:
so the original goal is to reuse VariableSavable for both tree variable and also fertile stats. which is another resources required for training.  https://github.com/tensorflow/tensorflow/pull/22418/files#diff-56893bd7583ce1f57ff27e7ebc64ea73R93 

Given the context, what would recommend here?

Reply:
Reusing VariableSaveable doesn't sound ideal, because each variable has it own way of saving and restoring, very likely not just restoring from one single Tensor. For this particular case it's probably fine but still prefer to name it TreeVariableSaveable to be clear.